ext/fs/std_fs.rs

```diff
@@ -769,6 +769,30 @@ fn stat_extra(
     Ok(info.dwVolumeSerialNumber as u64)
   }
 
+  use ntapi::ntioapi::FILE_ALL_INFORMATION;
+
+  unsafe fn query_file_information(
+    handle: winapi::shared::ntdef::HANDLE,
+  ) -> std::io::Result<FILE_ALL_INFORMATION> {
+    use ntapi::ntioapi::FileAllInformation;
+    use ntapi::ntioapi::NtQueryInformationFile;
+
+    let mut info = std::mem::MaybeUninit::<FILE_ALL_INFORMATION>::zeroed();
+    let status = NtQueryInformationFile(
+      handle,
+      std::ptr::null_mut(),
+      info.as_mut_ptr() as *mut _,
+      std::mem::size_of::<FILE_ALL_INFORMATION>() as _,
+      FileAllInformation,
+    );
+
+    if status < 0 {
+      return Err(std::io::Error::last_os_error());
+    }
+
+    Ok(info.assume_init())
+  }
```

Comment:
Does all this make everything that uses stat slower? (ex. `Deno.statSync`?) We should probably optimize stuff like `fs.existsSync` to not bother with all this stuff in case it's using it.

Reply:
Yeah, exists should be its own op. Here we can remove `get_dev`; it becomes redundant

Reply:
It seems this hasn't been addressed, so there's still perf penalty for APIs that don't need it

Reply:
Opened a PR for that: https://github.com/denoland/deno/pull/24613